examples/plot_post_stack_inversion.py

```diff
@@ -0,0 +1,120 @@
+from pylops.basicoperators import Transpose
+from pylops.avo.poststack import PoststackLinearModelling
+from pylops_mpi import MPIBlockDiag, DistributedArray
+
+plt.close("all")
+rank = MPI.COMM_WORLD.Get_rank()
+size = MPI.COMM_WORLD.Get_size()
+
+###############################################################################
+# Let's start by defining all of the parameters required by the
+# ``pylops.avo.poststack.PoststackLinearModelling`` operator.
+
+# Model
+model = np.load("../testdata/avo/poststack_model.npz")
+x, z, m = model['x'], model['z'], np.log(model['model'])
+
+# Making m a 3D model
+ny_i = 20  # size of model in y direction for rank i
+y = np.arange(ny_i)
+m3d_i = np.tile(m[:, :, np.newaxis], (1, 1, ny_i)).transpose((2, 1, 0))
+ny_i, nx, nz = m3d_i.shape
+
+# Size of y at all ranks
+ny = MPI.COMM_WORLD.allreduce(ny_i)
+
+# Smooth model
+nsmoothy, nsmoothx, nsmoothz = 5, 30, 20
+mback3d_i = filtfilt(np.ones(nsmoothy) / float(nsmoothy), 1, m3d_i, axis=0)
+mback3d_i = filtfilt(np.ones(nsmoothx) / float(nsmoothx), 1, mback3d_i, axis=1)
+mback3d_i = filtfilt(np.ones(nsmoothz) / float(nsmoothz), 1, mback3d_i, axis=2)
+
+# Wavelet
+dt = 0.004
+t0 = np.arange(nz) * dt
+ntwav = 41
+wav = ricker(t0[:ntwav // 2 + 1], 15)[0]
+
+# Collecting all the m3d and mback3d at all ranks
+m3d = np.concatenate(MPI.COMM_WORLD.allgather(m3d_i))
+mback3d = np.concatenate(MPI.COMM_WORLD.allgather(mback3d_i))
+
+###############################################################################
+# We now create the linear operator version of
+# ``pylops.avo.poststack.PostStackLinearModelling`` at each rank to model a
+# subset of the data along the y-axis. Such operators are passes
+# to the ``pylops_mpi.MPIBlockDiag`` operator, which is then used to perform
+# the different forward operations of each individual operator
+# at different ranks to compute the overall data. Note that to simplify the
+# handling of the model and data, we split and distribute the first axis,
+# and use ``pylops.basicoperators.Transpose`` to rearrange the model and data
+# in the form required by the ``pylops.avo.poststack.PostStackLinearModelling``
+# operator
+
+# Create flattened model data
+m3d_dist = DistributedArray(global_shape=ny * nx * nz)
+m3d_dist[:] = m3d_i.flatten()
+
+# LinearOperator PostStackLinearModelling
+PPop = PoststackLinearModelling(wav, nt0=nz, spatdims=(ny_i, nx))
+Top = Transpose((ny_i, nx, nz), (2, 0, 1))
+BDiag = MPIBlockDiag(ops=[Top.H @ PPop @ Top, ])
+
+# Data
+d_dist = BDiag @ m3d_dist
+d_local = d_dist.local_array.reshape((ny_i, nx, nz))
+d = d_dist.asarray().reshape((ny, nx, nz))
+
+if rank == 0:
+    # Check the distributed implementation gives the same result
+    # as the one running only on rank0
+    PPop0 = PoststackLinearModelling(wav, nt0=nz, spatdims=(ny, nx))
+    d0 = (PPop0 @ m3d.transpose(2, 0, 1)).transpose(1, 2, 0)
+
+    # Check the two distributed implementations give the same result
+    print('Distr == Local', np.allclose(d, d0))
```

Comment:
@mrava87 should we keep this or remove this?

Reply:
I would keep it, as a way to show how you can equivalently do it in pylops (for small 3d models) and that you get the same result :)